src/main/java/net/minecraftforge/fml/common/registry/ObjectHolderRef.java

```diff
@@ -44,15 +41,20 @@
 class ObjectHolderRef {
     private Field field;
     private ResourceLocation injectedObject;
-    private boolean isBlock;
-    private boolean isItem;
+    private boolean isValid;
+    private IForgeRegistry registry;
 
 
     ObjectHolderRef(Field field, ResourceLocation injectedObject, boolean extractFromExistingValues)
     {
+        Class<?> type = field.getType();
+        while (IForgeRegistryEntry.class.isAssignableFrom(type) && registry == null) {
+            registry = PersistentRegistryManager.findRegistryByType((Class<IForgeRegistryEntry>)type);
+            type = type.getSuperclass();
```

Comment:
Theoretically, it seems to, but attempting to test it for real highlighted an important issue, ObjectHolderRef objects are created before PreInit, and excluded from being applied if the registry doesn't exist prior to PreInit.

Without adding a new event specifically to create registries, I'm not sure I can alter this to support 3rd party registries :/

Alternatively I can alter the process to do registry resolution to after preInit.